Userland/Applications/FileManager/FileUtils.cpp

```diff
@@ -1,49 +1,22 @@
 /*
  * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2021, Sam Atkins <atkinssj@gmail.com>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #include "FileUtils.h"
+#include "FileOperationProgressWidget.h"
 #include <AK/LexicalPath.h>
-#include <LibCore/DirIterator.h>
 #include <LibCore/File.h>
 #include <LibGUI/MessageBox.h>
-#include <sys/stat.h>
 #include <unistd.h>
 
-namespace FileUtils {
+namespace FileManager {
 
-void delete_path(const String& path, GUI::Window* parent_window)
-{
-    struct stat st;
-    if (lstat(path.characters(), &st)) {
-        GUI::MessageBox::show(parent_window,
-            String::formatted("lstat({}) failed: {}", path, strerror(errno)),
-            "Delete failed",
-            GUI::MessageBox::Type::Error);
-    }
-
-    bool is_directory = S_ISDIR(st.st_mode);
-    auto result = Core::File::remove(path, Core::File::RecursionMode::Allowed, false);
-
-    if (result.is_error()) {
-        auto& error = result.error();
-        if (is_directory) {
-            GUI::MessageBox::show(parent_window,
-                String::formatted("Failed to delete directory \"{}\": {}", error.file, error.error_code),
-                "Delete failed",
-                GUI::MessageBox::Type::Error);
-        } else {
-            GUI::MessageBox::show(parent_window,
-                String::formatted("Failed to delete file \"{}\": {}", error.file, error.error_code),
-                "Delete failed",
-                GUI::MessageBox::Type::Error);
-        }
-    }
-}
+HashTable<RefPtr<GUI::Window>> file_operation_windows;
```

Comment:
This should probably be a `HashTable<NonnullRefPtr<GUI::Window>>`, as I don't think you want to store a null `RefPtr` in there.

Reply:
That makes sense. I'll have to look through code more carefully when moving it around in future, I basically just left it as it was before.

Reply:
If you just refactored that, it's not your fault of course (and also not really your responsibility). I just pointed it out because I saw it, and since you're working on it, it makes sense for you to quickly fix it. Code review shouldn't be about blaming people, it should be about producing the best possible code. :^)

Reply:
Sure! And you didn't make me feel that, don't worry. :^) I've only ever worked on my own code alone, so I felt nervous initially about having people critique what I've done, but it's been really positive. I really appreciate the time you and the others put into reviewing.